pkg/ctl/create/cluster.go

```diff
@@ -424,6 +417,75 @@ func doCreateCluster(cmd *cmdutils.Cmd, ngFilter *filter.NodeGroupFilter, params
 	return nil
 }
 
+// installKarpenter prepares the environment for Karpenter, by creating the following resources:
```

Comment:
Trying to add unit tests fro this next.

Reply:
@nikimanoledaki So I looked at this now for a long time, but in order to unit test this, I have to refactor a LOT of things.

While I think that has a lot of merit, doing that in this Pr would be a nightmare to review. Maybe this could be added to the testing refactor? Or I could create a separate ticket to refactor the `ctl` package to be unit testable.